### PR TITLE
Function statement locations now only include the function header

### PR DESCRIPTION
**Pull request by tomwhatmore**

### CODE REVIEW

File: src/grammar.lalrpop [hunk 1]
```diff
@@ -96,9 +96,9 @@ ExternalFnArg: ExternalFnArg = {
 }
 
 StatementFn: UntypedStatement = {
-    <s:@L> <p:"pub"?> "fn" <n:VarName> "(" <a:Comma<FnArg>> ")" <return_annotation:("->" <Type>)?> "{" <b:Expr+> "}" <e:@L> => Statement::Fn {
+    <s:@L> <p:"pub"?> "fn" <n:VarName> "(" <a:Comma<FnArg>> ")" <return_annotation:("->" <Type>)?> <e:@L> "{" <b:Expr+> "}"  => Statement::Fn {
         doc: None,
-        location: location(s, e),
+        location: location(s, e - 1),
```
Comment (lpil) [inline src/grammar.lalrpop:101]: Why is it - 1 ?

Comment (tomwhatmore) [reply src/grammar.lalrpop:101]: It looks better (in my opinion)! One less char of underline.

`location(s, e) ` -> https://i.imgur.com/25oep7K.png
`location(s, e - 1) ` -> https://i.imgur.com/ffS3tf0.png

Comment (lpil) [reply src/grammar.lalrpop:101]: What does it look like with a return annotation?

Comment (tomwhatmore) [reply src/grammar.lalrpop:101]: Same thing, one extra char of whitespace without the - 1.

`location(s, e) ` -> https://i.imgur.com/N2oOabR.png
`location(s, e - 1) ` -> https://i.imgur.com/8recJ8u.png

Comment (lpil) [reply src/grammar.lalrpop:101]: Nice! Thanks for sharing that extra info!

Comment (tomwhatmore) [reply src/grammar.lalrpop:101]: No problem! I should try and come up with a better workflow for this than manually uploading screenshots to imgur.

**Pull request merged**